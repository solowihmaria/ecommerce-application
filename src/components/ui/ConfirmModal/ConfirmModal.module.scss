@use '../../../assets/styles/constants.module' as c;

.modalOverlay {
    position: fixed;
    z-index: 1000;
    inset: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 1rem;
    backdrop-filter: blur(c.$blur5);
}

.modal {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    width: 100%;
    max-width: 30rem;
    padding: 2rem;
    border: 1px solid c.$color-text-primary;
    border-radius: 8px;
    background-color: c.$color-primary-dark;
    box-shadow: 0 4px 6px c.$opacity-5;
}

.modalTitle {
    margin-top: 0;
    margin-bottom: 1rem;
    font-size: 1.25rem;
    font-weight: 600;
    color: c.$color-text-primary;
    text-align: center;
}

.modalMessage {
    margin-bottom: 2rem;
    color: c.$color-text-primary;
    text-align: center;
}

.modalActions {
    display: flex;
    flex-wrap: wrap;
    gap: 1rem;
    justify-content: center;
}

@media (width <= 480px) {
    .modal {
        padding: 1.5rem;
    }

    .modalActions {
        flex-direction: column;
        width: 100%;
    }
}
