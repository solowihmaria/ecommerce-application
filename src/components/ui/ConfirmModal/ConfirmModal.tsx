import { Button } from '../../../components/ui/Button';
import styles from './ConfirmModal.module.scss';
import type { ConfirmModalProps } from './ConfirmModal.types';

export const ConfirmModal = ({
    isOpen,
    title,
    message,
    onConfirm,
    onCancel,
    confirmText = 'Confirm',
    cancelText = 'Cancel',
}: ConfirmModalProps) => {
    if (!isOpen) {
        return null;
    }

    return (
        <div className={styles.modalOverlay}>
            <div className={styles.modal}>
                <h3 className={styles.modalTitle}>{title}</h3>
                <p className={styles.modalMessage}>{message}</p>
                <div className={styles.modalActions}>
                    <Button variant="outline" onClick={onCancel}>
                        {cancelText}
                    </Button>
                    <Button variant="danger" onClick={onConfirm}>
                        {confirmText}
                    </Button>
                </div>
            </div>
        </div>
    );
};
