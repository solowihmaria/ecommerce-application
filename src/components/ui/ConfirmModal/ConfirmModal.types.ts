export interface ConfirmModalProps {
    title?: string;
    message: string;
    confirmText?: string;
    cancelText?: string;
    onConfirm: () => void;
    onCancel: () => void;
    isLoading?: boolean;
}
