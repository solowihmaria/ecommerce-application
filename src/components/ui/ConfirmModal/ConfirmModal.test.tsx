import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfirmModal } from './ConfirmModal';
import type { ConfirmModalProps } from './ConfirmModal.types';

describe('<ConfirmModal/> component', () => {
    const defaultProps: ConfirmModalProps = {
        isOpen: true,
        title: 'Delete Item',
        message: 'Are you sure you want to delete this?',
        confirmText: 'Yes, delete',
        cancelText: 'No, cancel',
        onConfirm: jest.fn(),
        onCancel: jest.fn(),
    };

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('does not render when isOpen is false', () => {
        render(<ConfirmModal {...defaultProps} isOpen={false} />);

        expect(screen.queryByText(defaultProps.title)).toBeNull();
        expect(screen.queryByText(defaultProps.message)).toBeNull();
    });

    test('renders correctly with all props', () => {
        render(<ConfirmModal {...defaultProps} />);
        const title = screen.getByText(defaultProps.title);
        const message = screen.getByText(defaultProps.message);
        const cancelButton = screen.getByRole('button', {
            name: defaultProps.cancelText,
        });
        const confirmButton = screen.getByRole('button', {
            name: defaultProps.confirmText,
        });

        expect(title).toBeInTheDocument();
        expect(message).toBeInTheDocument();
        expect(cancelButton).toHaveClass('button', 'outline');
        expect(confirmButton).toHaveClass('button', 'danger');

        expect(document.body).toMatchSnapshot();
    });

    test('calls onCancel when cancel button is clicked', () => {
        const onCancel = jest.fn();
        render(<ConfirmModal {...defaultProps} onCancel={onCancel} />);
        const cancelButton = screen.getByRole('button', {
            name: defaultProps.cancelText,
        });
        fireEvent.click(cancelButton);
        expect(onCancel).toHaveBeenCalledTimes(1);
    });

    test('calls onConfirm when confirm button is clicked', () => {
        const onConfirm = jest.fn();
        render(<ConfirmModal {...defaultProps} onConfirm={onConfirm} />);
        const confirmButton = screen.getByRole('button', {
            name: defaultProps.confirmText,
        });
        fireEvent.click(confirmButton);
        expect(onConfirm).toHaveBeenCalledTimes(1);
    });
});
