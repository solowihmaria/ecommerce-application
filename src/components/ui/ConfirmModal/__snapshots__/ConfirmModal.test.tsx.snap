// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmModal/> component renders correctly with all props 1`] = `
<body>
  <div>
    <div
      class="modalOverlay"
    >
      <div
        class="modal"
      >
        <h3
          class="modalTitle"
        >
          Delete Item
        </h3>
        <p
          class="modalMessage"
        >
          Are you sure you want to delete this?
        </p>
        <div
          class="modalActions"
        >
          <button
            class="button outline"
          >
            No, cancel
          </button>
          <button
            class="button danger"
          >
            Yes, delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
